src/compiler/checker.ts

```diff
@@ -16394,6 +16399,18 @@ namespace ts {
         }
 
         /**
+         * Sometimes, we have a decorator that could accept zero arguments,
+         * but is receiving too many arguments as part of the decorator invocation.
+         * In those cases, a user may have meant to *call* the expression before using it as a decorator.
+         */
+        function isPotentiallyUncalledDecorator(decorator: Decorator, signatures: Signature[]) {
+            return signatures.length && every(signatures, signature =>

```

Comment:
should you also check that the return type is callable/`any` to ensure your suggestion of calling it is warranted.

Reply:
`any` is accounted for - it's an uncalled function call. It also has no signatures.